Microsoft.Azure.WebJobs.Extensions.DurableTask.AzureManaged 1.6.0 has stale AzureBlobPayloads 1.23.3 transitive — causes NU1605 in latest Functions Worker apps

## Summary

`Microsoft.Azure.WebJobs.Extensions.DurableTask.AzureManaged 1.6.0` (the WebJobs counterpart auto-injected by `Microsoft.Azure.Functions.Worker.Sdk` when consumers reference `Microsoft.Azure.Functions.Worker.Extensions.DurableTask.AzureManaged 1.6.0`) brings in `Microsoft.DurableTask.Extensions.AzureBlobPayloads 1.23.3`. That transitive's floor on `Microsoft.Extensions.DependencyInjection.Abstractions >= 8.0.2` conflicts with the floor demanded elsewhere in the same restore graph (`>= 9.0.1` via `Microsoft.NET.Sdk.Functions 4.6.0 → Microsoft.Azure.WebJobs 3.0.45 → Microsoft.Extensions.Logging.Abstractions 9.0.1`), producing an unsuppressible `NU1605`.

`Microsoft.DurableTask.Extensions.AzureBlobPayloads` has shipped through `1.24.1` on NuGet — only `AzureManaged 1.6.0` is still pinned to `1.23.3`.

## Repro

Any isolated-worker Functions Orchestrator at the latest stable versions:

| Package | Version |
|---|---|
| `Microsoft.Azure.Functions.Worker.Sdk` | 2.0.7 |
| `Microsoft.Azure.Functions.Worker.Extensions.DurableTask` | 1.16.3 |
| `Microsoft.Azure.Functions.Worker.Extensions.DurableTask.AzureManaged` | 1.6.0 |

…fails build with:

~~~
error NU1605: Detected package downgrade: Microsoft.Extensions.DependencyInjection.Abstractions from 9.0.1 to 8.0.2.
- Microsoft.Azure.Functions.Worker.Extensions
    -> Microsoft.NET.Sdk.Functions 4.6.0
    -> Microsoft.Azure.WebJobs 3.0.45
    -> Microsoft.Extensions.Logging.Abstractions 9.0.1
    -> Microsoft.Extensions.DependencyInjection.Abstractions (>= 9.0.1)
- Microsoft.Azure.Functions.Worker.Extensions
    -> Microsoft.Azure.WebJobs.Extensions.DurableTask.AzureManaged 1.6.0
    -> Microsoft.DurableTask.Extensions.AzureBlobPayloads 1.23.3
    -> Microsoft.Extensions.DependencyInjection.Abstractions (>= 8.0.2)
~~~

The error fires in `obj/.../WorkerExtensions/WorkerExtensions.csproj` (the metadata sub-build the Worker SDK auto-generates).

## Expected

`Microsoft.Azure.WebJobs.Extensions.DurableTask.AzureManaged` should reference a `Microsoft.DurableTask.Extensions.AzureBlobPayloads` version whose `Microsoft.Extensions.DependencyInjection.Abstractions` floor is compatible with what the latest `Microsoft.NET.Sdk.Functions` brings in (>= 9.0.1).

## Suggested fix

Bump the `AzureManaged` package's transitive on `Microsoft.DurableTask.Extensions.AzureBlobPayloads` from `1.23.3` to `>= 1.24.0` (or whichever release first widens the DI.Abstractions floor to `>= 9.x`) and ship as `1.6.1` / `1.7.0`.

This affects only this single dependency edge — the surface area of the change should be minimal.

## Why this hits everyone using AzureManaged on Worker isolated

- `1.6.0` is the latest published version.
- `Microsoft.Azure.Functions.Worker.Sdk 2.0.7` (latest) hardcodes `Microsoft.NET.Sdk.Functions 4.6.0` (latest) into the auto-generated `WorkerExtensions.csproj`, which transitively requires `Microsoft.Extensions.DependencyInjection.Abstractions >= 9.0.1`.
- The auto-generated csproj does **not** inherit `Directory.Packages.props` or `Directory.Build.props`, so consumers cannot pin from the parent.
- `NU1605` is an error (not a warning) for SDK-style projects and cannot be suppressed via `NoWarn`.
- Consumers therefore have no clean way to fix this without post-processing the generated csproj or pinning to older versions of the SDK/Functions packages.

## Workaround (consumers)

Patch the generated `WorkerExtensions.csproj` between Worker SDK targets (see linked issue at [Azure/azure-functions-dotnet-worker](https://github.com/Azure/azure-functions-dotnet-worker/issues/3388)).

## Environment

- `Microsoft.Azure.Functions.Worker.Extensions.DurableTask.AzureManaged` 1.6.0
- `Microsoft.Azure.Functions.Worker.Sdk` 2.0.7
- `Microsoft.NET.Sdk.Functions` 4.6.0 (auto-injected)
- .NET SDK 10.0.203, target `net10.0` (Worker SDK sub-build target `net8.0`)
- Windows 11 Pro

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
+- Bump `Microsoft.Extensions.DependencyInjection.Abstractions` from 8.0.2 to 9.0.1 (and `Microsoft.Bcl.AsyncInterfaces` from 8.0.0 to 9.0.1, which the former transitively floors at 9.0.1) to align with the floor declared by `Microsoft.Azure.WebJobs 3.0.45 -> Microsoft.Extensions.Logging.Abstractions 9.0.1`. Fixes NU1605 in downstream Azure Functions Worker isolated apps consuming `Microsoft.DurableTask.Extensions.AzureBlobPayloads` ([Azure/azure-functions-durable-extension#3433](https://github.com/Azure/azure-functions-durable-extension/issues/3433)).
 - Validate explicit `UseWorkItemFilters(filters)` filter names against the worker's `DurableTaskRegistry`. Filters that reference an orchestration, activity, or entity name not registered with the worker now throw `OptionsValidationException` at worker startup instead of silently waiting for work items that will never arrive. No customer-side validation call is required. ([#719](https://github.com/microsoft/durabletask-dotnet/pull/719))
 
 ## 1.24.1

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -9,16 +9,18 @@
 
   <!-- Microsoft.Extensions.* Packages
        IMPORTANT: These assemblies are part of the .NET shared framework and load into the
-       Azure Functions host process, which runs on net8.0. Package versions MUST stay at 8.x
-       so the assembly versions (8.0.0.0) match what the host provides. Using 10.x causes
-       FileNotFoundException at runtime because the host cannot satisfy Version=10.0.0.0.
-       DO NOT upgrade to 10.x+ until the Azure Functions host itself targets .NET 10.
-       See: https://github.com/Azure/azure-functions-host (WebJobs.Script.csproj -> net8.0) -->
+       Azure Functions host process. Historically the host ran on net8.0 and required the
+       package versions to stay at 8.x so the assembly versions (8.0.0.0) matched what the
+       host provided; using 10.x caused FileNotFoundException at runtime.
+       As of azure-functions-host v4.1049.x the host targets net10.0 and ships 9.x/10.x
+       shared-framework assemblies via runtimeassemblies-relaxed.json (minorMatchOrLower),
+       so 9.x package versions are now safe.
+       See: https://github.com/Azure/azure-functions-host (WebJobs.Script.csproj -> net10.0) -->
   <ItemGroup>
     <PackageVersion Include="Microsoft.Extensions.Caching.Memory" Version="8.0.1" />
     <PackageVersion Include="Microsoft.Extensions.Configuration.Abstractions" Version="8.0.0" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="10.0.2" />
-    <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.2" />
+    <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="9.0.1" />
     <PackageVersion Include="Microsoft.Extensions.Hosting" Version="8.0.1" />
     <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="8.0.1" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="8.0.3" />
@@ -88,7 +90,7 @@
     <PackageVersion Include="DotNext" Version="4.13.1" Condition="'$(TargetFramework)' != 'net8.0' AND '$(TargetFramework)' != 'net10.0'" />
     <PackageVersion Include="DotNext" Version="5.19.0" Condition="'$(TargetFramework)' == 'net8.0' OR '$(TargetFramework)' == 'net10.0'" />
     <PackageVersion Include="Microsoft.SourceLink.GitHub" Version="8.0.0" />
-    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0" />
+    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="9.0.1" />
     <PackageVersion Include="System.Linq.Async" Version="6.0.1" />
     <PackageVersion Include="System.Text.Json" Version="8.0.6" />
     <PackageVersion Include="System.Collections.Immutable" Version="8.0.0" />

